src/analysis/buildTypeWeights.ts

```diff
@@ -8,19 +42,168 @@ import { GraphQLSchema } from 'graphql/type/schema';
+                            // TODO: Get variable value and return
+                            // const multiplier: number =
+                            // return result[listType.name.toLowerCase()].weight * multiplier;
+                            throw new Error(
+                                'ERROR: buildTypeWeights Variable arge values not supported;'
+                            );
+                        }
+                    }
+
+                    // FIXME: The list is unbounded. Return the object weight for
+                    throw new Error(
+                        `ERROR: buildTypeWeights: Unbouned list complexity not supported. Query results should be limited with ${KEYWORDS}`
+                    );
+                };
+            }
+        });
+
+        // if the field is a scalar or an enum set weight accordingly
+        if (isScalarType(resolveType) || isEnumType(resolveType)) {
+            result.query.fields[field] = typeWeights.scalar || DEFAULT_SCALAR_WEIGHT;
+        }
+    });
+    return result;
+}
+
+/**
+ * Parses all types in the provided schema object excempt for Query, Mutation
+ * and built in types that begin with '__' and outputs a new TypeWeightObject
+ * @param schema
+ * @param typeWeightObject
+ * @param typeWeights
+ * @returns
+ */
+function parseTypes(
+    schema: GraphQLSchema,
+    typeWeightObject: TypeWeightObject,
+    typeWeights: TypeWeightConfig
+): TypeWeightObject {
+    const typeMap: ObjMap<GraphQLNamedType> = schema.getTypeMap();
+
+    const result: TypeWeightObject = { ...typeWeightObject };
+
+    // Handle Object, Interface, Enum and Union types
+    Object.keys(typeMap).forEach((type) => {
+        const typeName = type.toLowerCase();
+
+        const currentType: GraphQLNamedType = typeMap[type];
+        // Get all types that aren't Query or Mutation or a built in type that starts with '__'
+        if (type !== 'Query' && type !== 'Mutation' && !type.startsWith('__')) {
+            if (isObjectType(currentType) || isInterfaceType(currentType)) {
+                // Add the type and it's associated fields to the result
+                result[typeName] = {
+                    fields: {},
+                    weight: typeWeights.object || DEFAULT_OBJECT_WEIGHT,
+                };
+
+                const fields = currentType.getFields();
+
+                Object.keys(fields).forEach((field: string) => {
+                    const fieldType: GraphQLOutputType = fields[field].type;
+
+                    // Only scalars are considered here any other types should be references from the top level of the type weight object.
+                    if (
+                        isScalarType(fieldType) ||
+                        (isNonNullType(fieldType) && isScalarType(fieldType.ofType))
+                    ) {
+                        result[typeName].fields[field] =
+                            typeWeights.scalar || DEFAULT_SCALAR_WEIGHT;
```

Comment:
Based on line `219` in this file I think the `DEFAULT_SCALAR_WEIGHT` is redundant as the `scalar` should be non-null.

Reply:
This is done on line `71` of the `parseQuery` function. Not really a big deal.